accepted/00083-containerized-proxy.md

```diff
@@ -0,0 +1,87 @@
+- Feature Name: Deliver Proxy as a container
+- Start Date: 2021-10-11
+
+# Summary
+[summary]: #summary
+
+Deliver Proxy as a container application.
+
+# Motivation
+[motivation]: #motivation
+
+There are deployment scenarios in which adding a new Virtual Machine to implement Proxy functionality is cumbersome or impossible, while leveraging an existing container runtime cold be an easier option, especially in highly distributed ("Edge") environments.
+
+Having the option of deploying Proxy functionality as a container can be easier to operate (eg. deployment, independence from underlying OS, keeping up to date) and more flexible (eg. implementing active/passive HA by leveraging an existing Kubernetes environment).
+
+It is believed implementing Proxy containerization will surface a subset of common problems and solutions also useful to implement a containerized Server in future.
+
+# Detailed design
+[design]: #detailed-design
+
+## Delivery
+  - Proxy is delivered as one container
+    - Updates are delivered via new images on a public Registry
+  - Documentation is produced to describe how this container can be run on popular container runtimes: [docker](https://docs.docker.com/engine/reference/commandline/cli/) and [podman](https://podman.io/)
+  - A minimal [Helm](https://helm.sh/) chart is delivered and tested against [k3s](https://k3s.io/)
+
+## Onboarding
+  - Containerized Proxy is not a managed client, does not run `salt-minion`, is not registered in the usual way
+  - A new UI page (and API endpoint) allows to add containerized Proxies by specifying their FQDN
+    - After confirmation a System entry (in the System list) is created 
+      - Normal client actions (package management, configuration file management...) will not be available on that System
+      - Proxy-specific Server UI/API functionality, like showing the list of attached clients, will be available like on regular Proxies
+    - for each such System, the Server generates a tarball ("configuration directory") that can be downloaded
+    - Proxy container needs the configuration directory to be mounted as a volume. After start, it is fully connected and operational
+
+## Operating
+  - Container will expose Salt (4505/4506) and https (443) ports
+  - Container will mount volumes:
+    - one with the configuration directory
```

Comment:
Cobbler PXE files are considered as a configuration directory? (`/srv/tftpboot`)

Reply:
Technically they shouldn't be stored on the host as the Squid Proxy currently serves that. Thus we would need an ephemeral volume for such a thing in my eyes.

Reply:
Then each time cobbler container or proxy container is started, cobbler sync would need to be called on SUMA, right?

Also squid proxy does not serve grub and pxe configs and kernel, initrds as they are using tftp (extept UEFI http, or course). Or did you have something else in mind?

Reply:
1. `cobbler sync` needs to be called each time we start the container - Yes and No the problem is that I hadn't have time to improve this. The sync is currently slow. Thus we have https://github.com/uyuni-project/uyuni-rfc/pull/44 . With the code of today: Yes - which is stupid because the content doesn't change and we sync to ALL proxies then AND Cobbler regenerates everything.
2. Yes TFTP is another matter and to be viewed separately from my comment.